modules/core/src/core/animation-loop.js

```diff
@@ -214,16 +211,7 @@ export default class AnimationLoop {
   }
 
   toDataURL() {
-    if (this._canvasDataURLPromise) {
-      return this._canvasDataURLPromise;
-    }
-
-    this.setNeedsRedraw('getCanvasDataUrl');
```

Comment:
Is dropping this `needsRedraw` hint intentional?

Reply:
So here's the thing... As far as I can tell that flag doesn't actually do anything. @ibgreen is the `needsRedraw` flag meant to just be informational?

Reply:
The needsRedraw flag is part of a redraw system that spans many classes in deck.gl and luma.gl.

The idea is that the `onRender` callback can check this flag and a number of other similar flags and determine if a redraw is needed.

By cutting it here, it would appear that there is no way for the application loop to determine that it has to actually draw in this case.

Reply:
That's what I assumed, but I don't see it being checked anywhere. Did I miss it?

Reply:
Well deck.gl will certainly need to start checking it now that screen capture is async. Since that is a recent change, deck.gl may not yet have been updated.

Reply:
Oh sorry I missed that it's the "application loop" that's determining if it needs to draw. So it is only informational from the point of view of `AnimationLoop`. I'll add it back to `toDataURL` and also to `waitForRender`, so those two can indicate they need a redraw.

Reply:
Thanks. 

I think the fact that this wasn't clear means is a good indicator that it is also worth checking the docs (and maybe code comments) to see if we are explaining this in a way that gives the user a reasonable chance of understanding the intentions behind these features.